### PR TITLE
Adapt spdx-analyzer to new path structure

### LINKED ISSUE #384: Fix SPDX analyzer
Due to #383 spdx analyzer has to be updated to the new data structure (path now is an array of structures and not a string)

Depends on #378

### PR DESCRIPTION
**Pull request by senaingrid**
Fix #384 

<br>

Waiting for approval to push these changes into #378

### CODE REVIEW

File: modules/analyzers/spdx-analyzer/spdxanalyzer/__main__.py [hunk 1]
```diff
@@ -36,6 +36,19 @@ def __stringify(t):
...
+    def __get_last_path_info(fnode):
+        """
+        Returns the last pathInfo added to the file node
+        """
+        if not isinstance(fnode, FileNode):
+            logging.error("'{}' is not an instance of FileNode.".format(fnode))
+            sys.exit(3)
+
+        i = len(fnode.paths)
+        test = fnode.paths[i-1]
```
Comment (GiasemiSh) [inline modules/analyzers/spdx-analyzer/spdxanalyzer/__main__.py]: Is this line needed?

File: modules/analyzers/spdx-analyzer/spdxanalyzer/__main__.py [hunk 2]
```diff
@@ -36,6 +36,19 @@ def __stringify(t):
...
+        return fnode.paths[i-1]
```
Comment (GiasemiSh) [inline modules/analyzers/spdx-analyzer/spdxanalyzer/__main__.py]: Also check if the filenode is valid first. To be valid it needs to have at least one path. If the length of the paths is zero then an error message should be returned or if the pathInfo.Path is nil.

File: modules/analyzers/spdx-analyzer/spdxanalyzer/__main__.py [hunk 3]
```diff
@@ -78,12 +91,14 @@ def _process_filenodes(self):
         stream_resp = self.cserv.GetFileNode(query_node)
 
         for node in stream_resp:
-            logging.info("Analyze node {}".format(node.path))
+            logging.info("Starting to analyze nodes ...")
```
Comment (GiasemiSh) [inline modules/analyzers/spdx-analyzer/spdxanalyzer/__main__.py]: Better move this log outside the loop

File: modules/analyzers/spdx-analyzer/spdxanalyzer/__main__.py (hunk 1 above)
Comment (senaingrid) [reply modules/analyzers/spdx-analyzer/spdxanalyzer/__main__.py]: Not at all.... my bad!!!!!!!!!

**Comment by GiasemiSh** [comment]
Merged into #383

**Pull request closed without merging**